### ISSUE TITLE
Cannot programmatically resize on Wayland since 0.28.0

### ISSUE THREAD

**Issue by torokati44** [rust-windowing/winit#2753]
Take this simple application (based on this example: https://github.com/parasyte/pixels/tree/main/examples/minimal-winit):
https://gist.github.com/torokati44/e21a9576966395d7affd806f5c236a75

Compile it with `Cargo.toml`:
```toml
[package]
name = "winitbug"
version = "0.1.0"
edition = "2021"

[dependencies]
winit="0.27.5" # !!!
winit_input_helper="0.13.0" # !!!

pixels="0.12.0"
error-iter="0.4.1"
log="0.4.17"
env_logger="0.10.0"
```

When the mouse cursor is moved into the top left 20x20 area, the window resizes itself to 100x100, and the box will only bounce within this smaller window.

This works as intended with the versions given above.

However, when upgrading to:
```toml
[dependencies]
winit="0.28.0"
winit_input_helper="0.14.0"
```

I get this instead:

![image](https://user-images.githubusercontent.com/288816/228911630-e026aa7a-e5e0-45ed-bb3c-d81fc994fff9.png)

Manually resizing the window (or even just taking focus away from it) instantly fixes the surface size.

Both dependency versions work as intended on XWayland.

System info:

Operating System: Fedora Linux 37
KDE Plasma Version: 5.27.3
KDE Frameworks Version: 5.104.0
Qt Version: 5.15.8
Kernel Version: 6.2.7-200.fc37.x86_64 (64-bit)
Graphics Platform: Wayland
Processors: 32 × AMD Ryzen 9 5950X 16-Core Processor
Memory: 31,2 GiB of RAM
Graphics Processor: AMD Radeon RX 6600 XT

**Comment by torokati44** [comment]
Possibly related: https://github.com/rust-windowing/winit/issues/2499

**Comment by kchibisov** [comment]
See https://github.com/rust-windowing/winit/pull/2676 .

**Comment by kchibisov** [comment]
Fixed in 2496098
